scene/gui/text_edit.cpp

```diff
@@ -5691,6 +5825,43 @@ void TextEdit::_post_shift_selection() {
 	selection.selecting_text = true;
 }
 
+bool TextEdit::_is_caret_inside_selection(bool p_edge) const {
+	Point2i pos = get_line_column_at_pos(get_local_mouse_pos());
+	int row = pos.y;
+	int col = pos.x;
+	int num_rows_selection = selection.to_line - selection.from_line + 1;
+	if (p_edge) {
+		if (num_rows_selection == 1) {
+			if (col >= selection.from_column && col <= selection.to_column && row == selection.from_line) {
+				return true;
+			}
+		} else if (num_rows_selection == 2) {
+			if ((col >= selection.from_column && row == selection.from_line) || (col <= selection.to_column && row == selection.to_line)) {
+				return true;
+			}
+		} else {
+			if ((col >= selection.from_column && row == selection.from_line) || (col <= selection.to_column && row == selection.to_line) || (row > selection.from_line && row < selection.to_line)) {
+				return true;
+			}
+		}
+	} else {
+		if (num_rows_selection == 1) {
+			if (col > selection.from_column && col < selection.to_column && row == selection.from_line) {
+				return true;
+			}
+		} else if (num_rows_selection == 2) {
+			if ((col > selection.from_column && row == selection.from_line) || (col < selection.to_column && row == selection.to_line)) {
+				return true;
+			}
+		} else {
+			if ((col > selection.from_column && row == selection.from_line) || (col < selection.to_column && row == selection.to_line) || (row > selection.from_line && row < selection.to_line)) {
+				return true;
+			}
+		}
+	}
+	return false;
```

Comment:
This seems overly complex, would something like the following work?
```c++
if (p_edge) {
  if ((row == selection.from_line && col == selection.from_column) || (row == selection.to_line && col == selection.to_column)) {
    return true
  }
}
return (row > selection.from_line && row <= selection.to_line && (row > selection.from_line || col >= selection.from_column) && (row < selection.to_line || col < selection.to_column))
```

Reply:
I've used the first part of your code to simplify the current code.
But the second part

`return (row > selection.from_line && row <= selection.to_line && (row > selection.from_line || col >= selection.from_column) && (row < selection.to_line || col < selection.to_column))`

is not correct, it does not handle all the possible cases.

Reply:
To be honest I didn't thoroughly analyze that line of code, I stopped at
`row > selection.from_line &&`
that exclude the first line of selection, so I assumed you misunderstood the meaning of the p_edge parameter and used my code.

Looking back at your code I now understand what you wanted to do, but to make it work some changes are needed:

`row > selection.from_line  -->  row >= selection.from_line`

and

`col >= selection.from_column  -->  col > selection.from_column`

Reply:
done